aamt_parse2.pm

```diff
@@ -645,42 +906,46 @@ sub pars_PublishSite
             if (@files > 0)
             {
                 &deployToSite($publishUrl, "$workingFolder/wwwroot", $userName, $userPWD, TRUE);
-            }                
+            } 
+	    
+               
             my $returnCode = -1;
             my $database;
             my $password;
             # DUMP MYSQL
             my $retries = 0;
-            while($returnCode != 0 && $retries < 5)
+            while($returnCode != 0 && $retries < 1)
             {
                 ilog_print(1,"\nBacking up database on: $dbHost...\n");
-                if ($DEBUG_MODE) { ilog_print(1,"\nmysqldump --single-transaction -h $dbHost -u $dbUser -p'$dbPass' $dbName > $workingFolder/mysqldump.sql\n"); }
+{ ilog_print(1,"mysqldump --single-transaction -h $dbHost -u $dbUser -p'dbPass' $dbName > workingFolder/mysqldump.sql\n\n"); }
+
                 `mysqldump --single-transaction -h $dbHost -u $dbUser -p'$dbPass' $dbName > $workingFolder/mysqldump.sql`;
 
                 $returnCode = $?;
                 $retries++;
-                if ($returnCode != 0 && $retries < 5)
+                if ($returnCode != 0 && $retries < 2)
                 {
                     ilog_print(1,"\nreturncode: $returnCode\n");
+		    ilog_print(1,"mysqldump --single-transaction -h $dbHost -u $dbUser -p'dbPass' $dbName > workingFolder/mysqldump.sql\n"); 
                     sleep 1;
                 }
             }
 
             # RESTORE SQL
-            ilog_print(1,"\nMoving database...\n");
+            ilog_print(1,"\nMoving database to: $rServer...\n");
+	    ilog_print(1, "mysql -u $rUsername -h $rServer -p'rPassword' $rDatabase < workingFolder/mysqldump.sql\n\n");	
             `mysql -u $rUsername -h $rServer -p'$rPassword' $rDatabase < $workingFolder/mysqldump.sql`;
             $returnCode = $?;
             if ($DEBUG_MODE) { ilog_print(1,"\nDEBUG: returncode: $returnCode\n"); }
             if ($returnCode != 0)
             {
-                ilog_print(1,"\n\nPublishing failed with return code: $returnCode\n");                
+                ilog_print(1,"\nMoving failed with return code: $returnCode\n");                
                 $strYesOrNo =" ";
+	        ilog_print(1, "mysql -u $rUsername -h $rServer -p'rPassword' $rDatabase < workingFolder/mysqldump.sql\n");	
+		ilog_print(1,"\nplease check and try moving databsae manually...\n");
                 while($strYesOrNo!~/^\s*[YynN]\s*$/)
                 {
-                    ilog_printf(1, "    Would you like to retry publishing the database for site [$strSiteName]? (Y/N):");
```

Comment:
You need to add a cmd line flag for running in unattended/non-interactive mode. If this flag is true, then you can skip retries, if it is false, then keep the current logic of asking the user for input.